src/main/java/com/untamedears/ItemExchange/utility/BlockUtility.java

```diff
@@ -0,0 +1,111 @@
+package com.untamedears.ItemExchange.utility;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.*;
+import org.bukkit.inventory.*;
+
+import com.untamedears.ItemExchange.DeprecatedMethods;
+import com.untamedears.ItemExchange.ItemExchangePlugin;
+
+public class BlockUtility {
+
+	public static final BlockFace[] CardinalFaces = new BlockFace[] {
+			BlockFace.NORTH,
+			BlockFace.EAST,
+			BlockFace.SOUTH,
+			BlockFace.WEST,
+			BlockFace.DOWN,
+			BlockFace.UP
+	};
+
+	public static BlockFace GetFacingDirection(Block block) {
+		byte data = DeprecatedMethods.getBlockMeta(block);
+		if ((data & 0x5) == 5)
+			return BlockFace.EAST;
+		else if((data & 0x4) == 4)
+			return BlockFace.WEST;
+		else if((data & 0x3) == 3)
+			return BlockFace.SOUTH;
+		else if((data & 0x2) == 2)
+			return BlockFace.NORTH;
+		else if((data & 0x1) == 1)
+			return BlockFace.UP;
+		else if((data & 0x0) == 0)
+			return BlockFace.DOWN;
+		else
+			return BlockFace.SELF;
+	}
+
+	public static BlockFace GetAttachedDirection(Block block) {
+		byte data = DeprecatedMethods.getBlockMeta(block);
+		if ((data & 0x5) == 5)
+			return BlockFace.UP;
+		else if((data & 0x4) == 4)
+			return BlockFace.NORTH;
+		else if((data & 0x3) == 3)
+			return BlockFace.SOUTH;
+		else if((data & 0x2) == 2)
+			return BlockFace.WEST;
+		else if((data & 0x1) == 1)
+			return BlockFace.EAST;
+		else if((data & 0x0) == 0)
+			return BlockFace.DOWN;
+		else
+			return BlockFace.SELF;
+	}
+
+	public static void PowerBlock(Block block, long time) {
+		// Ensure that timedelay exists
+		if (time < 0) time = 0;
```

Comment:
This should just throw an exception (like IllegalArgumentException) instead, toggling a button for 0 ticks doesnt make sense.

Reply:
Done.